ambry-rest/src/main/java/com.github.ambry.rest/NettyResponseChannel.java

```diff
@@ -474,6 +482,30 @@ private void handleChannelWriteFailure(Throwable cause, boolean propagateErrorIf
     }
   }
 
+  /**
+   * Logs the exception at the appropriate level.
+   * @param exception the {@link Exception} that has to be logged.
+   */
+  private void log(Exception exception) {
+    String uri = "unknown";
+    RestMethod restMethod = RestMethod.UNKNOWN;
+    if (request != null) {
+      uri = request.getUri();
+      restMethod = request.getRestMethod();
+    }
+    if (exception instanceof RestServiceException) {
+      RestServiceErrorCode errorCode = ((RestServiceException) exception).getErrorCode();
+      ResponseStatus responseStatus = ResponseStatus.getResponseStatus(errorCode);
+      if (responseStatus == ResponseStatus.InternalServerError) {
```

Comment:
They are logged at different logging levels. Since the default is usually INFO, the trace logging doesn't go into the log but the error handling does.